drivers/SmartThings/matter-lock/src/init.lua

```diff
@@ -426,19 +426,22 @@ end
 
 local function handle_reload_all_codes(driver, device, command)
   if (device:get_field(lock_utils.CHECKING_CREDENTIAL) == nil) then
+    lock_utils.lock_codes_event(device, {})
     device:set_field(lock_utils.CHECKING_CREDENTIAL, 1)
   else
     device.log.info(string.format("Delaying scanning since currently checking credential %d", device:get_field(lock_utils.CHECKING_CREDENTIAL)))
     device.thread:call_with_delay(2, function(t) handle_reload_all_codes(driver, device, command) end)
     return
   end
-  device:emit_event(capabilities.lockCodes.scanCodes("Scanning"))
-  device:send(
-    clusters.DoorLock.server.commands.GetCredentialStatus(
-      device, device:component_to_endpoint(command.component),
-      {credential_type = DoorLock.types.DlCredentialType.PIN, credential_index = device:get_field(lock_utils.CHECKING_CREDENTIAL)}
+  device.thread:call_with_delay(5, function(t)
```

Comment:
For my understanding, why was this delay added here? Was it discovered during testing that locks need additional time to respond to reloading the codes?

Reply:
When the driver is changed, all field values for COTA are initialized. (lock_utils.COTA_CRED)
So the driver creates a new pin for COTA. and I added 5 seconds because a new PIN had to be created and reloaded.

When proceeding without delay, there is an issue that the pin for COTA is duplicated because reloading is performed before the pin for COTA is created.